### ISSUE TITLE
react example

### ISSUE THREAD

**Issue by SunChero** [nats-io/nats.ws#93]
please provide an example on how to import this module from create-react-app application.

right now if you start a new project:
- create-react-app  myproject
- npm install nats.ws
- import { connect } from "nats.ws"

it will  generate an error:
```
./node_modules/nats.ws/nats.mjs 3499:17
Module parse failed: Unexpected token (3499:17)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
| 
| function defaultJsOptions(opts4) {
>   opts4 = opts4 ?? {};
|   return extend1({
|     apiPrefix: defaultPrefix,
```

thank you

**Comment by aricart** [comment]
you may need to update your tooling:
https://dev.to/uncomfortably_average/create-react-app-v3-3-0-released-3j7b

This issue may also have additional tips:
https://github.com/nats-io/nats.ws/issues/68

**Comment by aricart** [comment]
You can also try this:
https://github.com/facebook/create-react-app/issues/10702

**Comment by junpark-vcanus** [comment]
I had the same issue. My cra version was > 3.3.0(might be latest). 

I guessed there's no plugin or dependency for nullish cohe-stuff(??) in my CRA so I ran the ejection and checked webpack plugins and it included nullish stuff.
In ./node_modules/babel-preset-react-app/package.json, there's @babel/...nullish-coalescing-operator on dependencies.

I gave up dealing with babel and found some issues on this repo. I tried nats.ws@next as written in the issue #68 and Yeah! It passed the nullish stuff error but failed on optional chaining(?.). 

In nats.ws@next, the nullish stuff has been changed to ||. (opts4 = opts4 ?? {}; ====> opts4 = opts4 || {};)
Okay. So I changed the optional chaining parts in nats.js like

1. Line 3414: in class JetStreamClientImpl
from
```
const mh = opts?.headers || headers1();
```
to
```
const mh = (opts ? opts.headers : false) || headers1();
```

2. Line 6564: in somewhere
from
```
const { lang  } = nci?.protocol?.transport;
```
to
```
const { lang  } = (nci ? nci.protocol : false) ? nci.protocol.transport : false;
```

and it worked anyway(might be unexposed error with the change). And I found that there's been the similar issue on nats.deno(#454).

I'm kinda new to webpack & babel things and in a rush with the due date, so I couldn't checked any other possibilities. Sorry.

p.s. NATS is GREAT! Thanks for all of you, contributors!

**Comment by SunChero** [comment]
i made my way throught and got it working .. somehow without ejecting

ref:(https://github.com/SunChero/nats-ws-create-react-app)

basically what i did 
- clone this git repository:
- npm run build-esm this should generate the nats.js file ..
-  i had to go modify a few lines in the generated nats.js file

**Comment by aricart** [comment]
@SunChero glad you were able to unblock yourself. Using the latest, I was able to do this:

```javascript
import React, {Component} from 'react'
import {connect} from '../node_modules/nats.ws/lib/src/mod.js'
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.server = props.server;
    this.doConnect();
  }

  doConnect() {
    connect({servers: [this.server]})
      .then((nc) => {
        this.nc = nc;
        this.nc.jetstreamManager()
          .then((jsm) => {
            this.jsm = jsm;
            console.info(`connected jsm`);
            this.jsm.streams.add({name: "my", subjects: ["my.>"]})
              .then((info) => {
                console.info(info);
              })
          })
          .catch((err) => {
            console.error(`failed to connect to jetstream: ${err}`);
          })
      })
      .catch((err) => {
        console.error(`failed to connect: ${err}`);
      })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          Hello NATS.ws!!!
        </header>
      </div>
    );
  }
}

export default App;

```

**Comment by aricart** [comment]
For some reason react is selecting the `mjs` library when it should be picking the `cjs`, but even if selected, it will complain about the re-exported functionality. Pointing it to the entry point, works correctly. Note I didn't have to change anything in the code.

**Comment by aricart** [comment]
![image](https://user-images.githubusercontent.com/1032976/116432960-a9fef900-a80e-11eb-83af-4cc401ff48c5.png)

**Comment by SunChero** [comment]
thank you @aricart  this is great
on another note ... cloning the git repo and trying npm run build  was giving me a bunch of errors 
i had to create a tsconfig file and npm install other libraries like shx ..etc
i will check if its not something weired on my side and maybe create a new issue

thank you

**Comment by aricart** [comment]
to build the repo: 
```bash
npm run setup #(only the first time)
npm run build
```
